src/Logic/UsersHelper.php

```diff
@@ -0,0 +1,297 @@
+<?php
+
+namespace Newspack\MigrationTools\Logic;
+
+use Exception;
+use InvalidArgumentException;
+use Newspack\MigrationTools\Util\Log\CliLog;
+use Newspack\MigrationTools\Util\Log\FileLog;
+use WP_User;
+
+class UsersHelper {
+
+	/**
+	 * Get a user by an array that contains either the ID, user_email, user_login or user_nicename.
+	 *
+	 * @param array $data Array with one (or more) of the following keys: 'ID', 'user_email', 'user_login', 'user_nicename'.
+	 *
+	 * @return WP_User|bool The user if found, false otherwise.
+	 */
+	public static function get_user( array $data ): WP_User|bool {
+		$wp_user = false;
+		if ( ! empty( $data['ID'] ) ) {
+			$wp_user = get_user_by( 'ID', $data['ID'] );
+		} elseif ( ! empty( $data['user_email'] ) ) {
+			$wp_user = get_user_by( 'email', $data['user_email'] );
+		} elseif ( ! empty( $data['user_login'] ) ) {
+			$wp_user = get_user_by( 'login', $data['user_login'] );
+		} elseif ( ! empty( $data['user_nicename'] ) ) {
+			$wp_user = get_user_by( 'slug', $data['user_nicename'] );
+		}
+
+		return $wp_user ?? false;
+	}
+
+	/**
+	 * Get a username that is not in use from a desired username.
+	 *
+	 * If the desired username is in use, a counter will be appended to it until an unused username is found.
+	 *
+	 * @param string $desired_username Desired username.
+	 *
+	 * @return string An unused username.
+	 */
+	public static function get_unused_username( string $desired_username ): string {
+		$original_username = $desired_username;
+		$desired_username  = trim( $desired_username );
+		$max_length        = 60;
+		if ( strlen( $desired_username ) >= $max_length ) {
+			$desired_username = trim( mb_substr( $desired_username, 0, $max_length ) );
+			FileLog::get_logger( 'UsersHelper' )->warning(
+				sprintf(
+					'Shortened username to under %d chars from "%s" to "%s".',
+					$max_length,
+					$original_username,
+					$desired_username
+				)
+			);
+		}
+
+		$i = 0;
+		while ( username_exists( $desired_username ) ) {
+			$desired_username = self::append_number_and_ensure_length( $desired_username, ++$i, $max_length );
+		}
+		if ( $i > 0 ) {
+			CliLog::get_logger( 'UsersHelper' )->debug( sprintf( 'Generated username: %s', $desired_username ) );
+		}
+
+		return $desired_username;
+	}
+
+	/**
+	 * Get a nicename that is not in use from a desired nicename.
+	 *
+	 * If the desired nicename is in use, a counter will be appended to it until an unused nicename is found.
+	 *
+	 * @param string $desired_nicename Desired nicename.
+	 *
+	 * @return string An unused nicename.
+	 */
+	public static function get_unused_nicename( string $desired_nicename ): string {
+		$original_nicename = $desired_nicename;
+		$desired_nicename  = trim( $desired_nicename );
+		$max_length        = 50;
+		if ( strlen( $desired_nicename ) >= 50 ) {
```

Comment:
Love the changes! Just change `50` in the condition to `$max_length` so it becomes:

```php
if ( strlen( $desired_nicename ) >= $max_length ) {
```

Reply:
Good catch!